src/nvim/api/vim.c

```diff
@@ -232,8 +232,7 @@ void vim_change_directory(String dir, Error *err)
   }
 
   char string[MAXPATHL];
-  strncpy(string, dir.data, dir.size);
-  string[dir.size] = NUL;
+  xstrlcpy(string, dir.data, dir.size+1);
```

Comment:
Scratch that, if you look at the definition of `xstrlcpy()`,
```c
size_t xstrlcpy(char *restrict dst, const char *restrict src, size_t size) {
    size_t ret = strlen(src);
    // ...
}
```
If the concern is that `dir.data` may not be `NUL` terminated, then `xstrlcpy()` may not be used.